docs/copilot/mcp-servers.md

```diff
@@ -0,0 +1,211 @@
+
+_MCP servers_ are services that provide one or more tools to _MCP hosts_ like VS Code. For example, a file system MCP server might provide tools for reading, writing, or searching files and directories. MCP servers can run locally on your machine or be hosted remotely.
+
+VS Code supports MCP servers, allowing you to add them to your workspace and use the tools they provide in agent mode.
+
+> [!IMPORTANT]
+> MCP server support is currently in preview and only available in [VS Code Insiders](https://code.visualstudio.com/insiders).
+
+## Add an MCP server
+
+To add an MCP server to your workspace, follow these steps:
+
+1. Configure an MCP server in the `.vscode/mcp.json` file in your workspace to share configurations with project collaborators.
+
+    Select the **Add Server** button to add a template for a new server. VS Code provides IntelliSense for the MCP server configuration file.
+
+    The following example shows how to configure a Perplexity MCP server, where VS Code prompts you for the API key when the server starts. Learn more about the [Configuration format](#configuration-format).
+
+    ```json
+    // Example .vscode/mcp.json
+    {
+        // 💡 Inputs will be prompted on first server start,
+        //    then stored securely by VS Code.
+        "inputs": [
+            {
+                "type": "promptString",
+                "id": "perplexity-key",
+                "description": "Perplexity API Key"
+            },
+        ],
+        // Note, not "mcpServers"
+        "servers": {
+            // https://github.com/ppl-ai/modelcontextprotocol/
+            "Perplexity": {
+                "type": "stdio",
+                "command": "npx",
+                "args": [
+                    "-y",
+                    "@modelcontextprotocol/server-perplexity-ask"
+                ],
+                "env": {
+                    "PERPLEXITY_API_KEY": "${input:perplexity-key}"
+                }
+            }
+        }
+    }
+    ```
+
+1. Alternatively, specify the server in the `setting(mcp)` VS Code setting for a personalized MCP server list.
+
+    Add the server configuration to your [user settings](/docs/getstarted/personalize-vscode.md#configure-settings) to share MCP server configurations across all your workspaces. This is useful for personal servers or when you want to use the same server configuration in multiple projects.
+
+1. Enable auto-discovery of MCP servers in your workspace by setting the `setting(chat.mcp.discovery.enabled)` setting.
+
+    VS Code can automatically discover and reuse MCP servers that you have already defined in other tools, such as Claude Desktop and others.
+
+1. Run the **MCP: List Servers** command from the Command Palette to view and manage the configured MCP servers.
+
+## Configuration format
+
+Use the following JSON configuration format to define MCP servers.
```

Comment:
Somewhere in here we should link to https://code.visualstudio.com/docs/reference/variables-reference, and the `inputs` section should link to the input variables section on the page https://code.visualstudio.com/docs/reference/variables-reference#_input-variables